src/components/panel/panel.js

```diff
@@ -682,52 +709,77 @@ function MdPanelService($rootElement, $rootScope, $injector, $window) {
   /** @private @const */
   this._$window = $window;
 
+  /** @private {!Object<string, !MdPanelRef>} */
+  this._trackedPanels = {};
 
   /**
+   * @description
    * Default animations that can be used within the panel.
+   *
    * @type {enum}
    */
   this.animation = MdPanelAnimation.animation;
 
   /**
+   * @description
    * Possible values of xPosition for positioning the panel relative to
    * another element.
+   *
    * @type {enum}
    */
   this.xPosition = MdPanelPosition.xPosition;
 
   /**
+   * @description
    * Possible values of yPosition for positioning the panel relative to
    * another element.
+   *
    * @type {enum}
    */
   this.yPosition = MdPanelPosition.yPosition;
 }
 
 
 /**
+ * @description
  * Creates a panel with the specified options.
+ *
  * @param {!Object=} config Configuration object for the panel.
  * @returns {!MdPanelRef}
  */
 MdPanelService.prototype.create = function(config) {
-  var configSettings = config || {};
+  config = config || {};
+
+  // If the passed-in config contains an ID and the ID is within _trackedPanels,
+  // return the tracked panel.
+  if (angular.isDefined(config.id) &&
+      this._trackedPanels[config.id]) {
+        return this._trackedPanels[config.id];
+      }
 
+  // If no ID is set within the passed-in config, then create an arbitrary ID.
   this._config = {
+    id: config.id || 'panel_' + this._$injector.get('$mdUtil').nextUid(),
     scope: this._$rootScope.$new(true),
     attachTo: this._$rootElement
   };
-  angular.extend(this._config, this._defaultConfigOptions, configSettings);
+  angular.extend(this._config, this._defaultConfigOptions, config);
 
-  var instanceId = 'panel_' + this._$injector.get('$mdUtil').nextUid();
-  var instanceConfig = angular.extend({ id: instanceId }, this._config);
+  var panelRef = new MdPanelRef(this._config, this._$injector);
 
-  return new MdPanelRef(instanceConfig, this._$injector);
+  // Add the newly created panelRef to _trackedPanels if an ID was provided.
```

Comment:
I think it's better to add it always regardless of whether the user provided the id or you created it. It will help you later with the group logic and make it consistent across all uses of panel. Yes, you will never reuse it, but tracking will be easier.